alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetFindingsAction.kt

```diff
@@ -131,10 +137,34 @@ class TransportGetFindingsSearchAction @Inject constructor(
         }
     }
 
-    suspend fun search(searchSourceBuilder: SearchSourceBuilder): GetFindingsResponse {
+    suspend fun resolveFindingsIndexName(findingsRequest: GetFindingsRequest): String {
+        var indexName = ""
+
+        if (findingsRequest.findingIndexName.isNullOrEmpty() == false) {
+            // findingIndexName has highest priority, so use that if available
+            indexName = findingsRequest.findingIndexName
+        } else if (findingsRequest.monitorId != null) {
+            // second best is monitorId.
+            // We will use it to fetch monitor and then read indexName from dataSources field of monitor
+            withContext(Dispatchers.IO) {
+                val getMonitorRequest = GetMonitorRequest(findingsRequest.monitorId, -3L, RestRequest.Method.GET, null)
+                val getMonitorResponse: GetMonitorResponse =
+                    this@TransportGetFindingsSearchAction.client.suspendUntil {
+                        execute(GetMonitorAction.INSTANCE, getMonitorRequest, it)
+                    }
```

Comment:
Exception is caught on line 134 and sent back to caller.

Hmm falling back to default one might lead to spurious behavior. It would be better if user knows that GetMonitor Action failed and is retryable then to get back NOT_FOUND without knowing if the desired finding index was searched or not